aip/general/0193.md

```diff
@@ -34,401 +34,242 @@ is necessary, you **should** provide a link where a reader can get more
-    [`google.rpc.Code`][Code].
-  - *Example*: 8
-- `message string`
-  - A developer-facing error message, which **should** be in English.
-    Any user-facing error message **should** be located in the
-    LocalizedMessage payload.
-  - *Example message*: \"The zone \'us-east1-a\' does not have enough
-    resources available to fulfill the request. Try a different zone, or
-    try again later.\"
-- `details Object (repeated)`
-  - Additional text details about the error, which includes
-    `ErrorInfo`,`LocalizedMessage`, and so on.
-  - *Example*: Described in the [ErrorDetails](#error-details) section.
-
-
-**Important:** In the context of the `google.rpc.Status` protobuf
-message, the value of the field `code` is the numeric equivalent to the
-enum value chosen from [`google.rpc.Code`][Code]. For example, if
-[`INVALID_ARGUMENT`][InvalidArgument] is chosen, the value of `code`
-will be `3` (in the context of the Protobuf message,
-`google.rpc.Status`). However, when the error is expressed as JSON, as
-in the [sample above](#json-representation), the value of the field by
-the same name, `"code"`, is the HTTP status code equivalent of the
-selected `google.rpc.Code`. For example, the value of the field `"code"`
-in the JSON reply would be `400`.
-
-#### [ErrorDetails][details]
+### Status.code
 
-Google defines a set of [standard detail payloads][details] for error
-details, which cover most common needs for API errors.  Services
-**should** use these standard detail payloads when feasible.
+The `code` field is the status code, which **must** be the numeric value of
+one of the elements of the [`google.rpc.Code`][Code] enum.
 
-Structured details with machine-readable identifiers **must** be used so
-that users can write code against specific aspects of the error. Error
-message strings **may** change over time; however, if an error message
-does not have a machine-readable identifier *in addition to* the `code`
-string, changing the error message **must** be considered a
-backwards-incompatible change.
+For example, the value `5` is the numeric value of the `NOT_FOUND`
+enum element.
 
-The following section describes `ErrorInfo`, `LocalizedMessage`, and
-`Help` in more detail.
+### Status.details
 
-##### Uniqueness
+The `details` field allows messages with additional error information to
+be included in the error response, each packed in a `google.protobuf.Any`
+message.
+
+Google defines a set of [standard detail payloads][details] for error
+details, which cover most common needs for API errors. 
+Services **should** use these standard detail payloads when feasible.
 
 Each type of detail payload **must** be included at most once. For
 example, there **must not** be more than one [`BadRequest`][BadRequest]
 message in the `details`, but there **may** be a `BadRequest` and a
 [`PreconditionFailure`][PreconditionFailure].
 
-##### ErrorInfo payload
+Structured details with machine-readable identifiers **must** be used so
```

Comment:
I'm now considering removing this paragraph entirely. It's already covered later on, and the currnet "if an error message does not have a machine-readable identifier in addition to the message" effectively contradicts the requirement that we always provide an ErrorInfo.

Reply:
Yeah I'm a little unsure of the value this paragraph is now providing. Agreed that it adds some confusion.